llvm/lib/Target/RISCV/RISCVInstrInfoZvk.td

```diff
@@ -281,59 +275,44 @@ multiclass VPseudoVALU_V_S_NoMask_Zvk
 multiclass VPseudoVALU_VV_NoMask_Zvk {
   foreach m = MxListVF4 in {
     defvar mx = m.MX;
-    defvar WriteVIALUV_MX = !cast<SchedWrite>("WriteVIALUV_" # mx);
-    defvar ReadVIALUV_MX = !cast<SchedRead>("ReadVIALUV_" # mx);
-
     defm _VV : VPseudoTernaryNoMask_Zvk<m.vrclass, m.vrclass, m.vrclass, m>,
-               Sched<[WriteVIALUV_MX, ReadVIALUV_MX, ReadVIALUV_MX, ReadVMask]>;
+               SchedTernary<"WriteVIALUV", "ReadVIALUV", "ReadVIALUV", "ReadVIALUV", mx>;
   }
 }
 
 multiclass VPseudoVALU_VI_NoMask_Zvk {
   foreach m = MxListVF4 in {
     defvar mx = m.MX;
-    defvar WriteVIALUV_MX = !cast<SchedWrite>("WriteVIALUV_" # mx);
-    defvar ReadVIALUV_MX = !cast<SchedRead>("ReadVIALUV_" # mx);
-
     defm _VI : VPseudoTernaryNoMask_Zvk<m.vrclass, m.vrclass, uimm5, m>,
-               Sched<[WriteVIALUV_MX, ReadVIALUV_MX, ReadVIALUV_MX, ReadVMask]>;
+               SchedTernary<"WriteVIALUV", "ReadVIALUV", "ReadVIALUV", "ReadVIALUV", mx>;
   }
 }
 
 multiclass VPseudoVALU_VI_NoMaskTU_Zvk {
```

Comment:
It feels wrong to call this TU but not have a merge operand. Everyone else who uses `VPseudoBinaryNoMask` does not read merge op in RISCVInstrInfoVPseudos. Is TU a bad name here? vaeskf1 uses this class and according to the spec, it does not look like it reads vd it just does `set_velem(vd, EGW=128, i, w[3:0]);`

Reply:
That's a bug that instruction should have a merge op for tail undisturbed.

Reply:
I'll fix

Reply:
There is a merge op already. The naming is just bad. It's instantiating this multiclass

```
multiclass VPseudoBinaryNoMask<VReg RetClass,
                               VReg Op1Class,
                               DAGOperand Op2Class,
                               LMULInfo MInfo,
                               string Constraint = "",
                               int sew = 0> {
  let VLMul = MInfo.value, SEW=sew in {
    defvar suffix = !if(sew, "_" # MInfo.MX # "_E" # sew, "_" # MInfo.MX);
    def suffix : VPseudoBinaryNoMaskTU<RetClass, Op1Class, Op2Class,
                                       Constraint>;
  }
}
-
```

Not to be confused with the `class` that has the same name. The `VPseudoBinaryNoMaskTU` has a merge op.

Reply:
Got confused with the `class` of the same name. I have updated this to read merge op.